src/test/java/com/example/demo/controller/GreetingControllerTests.java

```diff
@@ -9,11 +10,12 @@
 
 class GreetingControllerTests {
 
+@InjectMocks
+GreetingService serviceMock;
```

Comment:
The mocks are supposed to be injected into a class that is under test. Not the one that is supposed to be mocked. This filed should be changed to: `private GreetingController greetingController;`